File: trunk/server/net/Protocols/EjectionResponse.cs
using ProtoBuf;
using System.Collections.Generic;
using System;

namespace protocol
{
    /// <summary>
    /// 出牌 协议:-21
    /// </summary>
	[Proto(value=-21,description="出牌")]
	[ProtoContract]
	public class EjectionResponse
	{
        /// <summary>
        ///  用户id
        /// </summary>
		[ProtoMember(3, IsRequired = false)]
		public string pid;
        /// <summary>
        ///  卡牌唯一id
        /// </summary>
		[ProtoMember(4, IsRequired = false)]
		public string uniqueId;
        /// <summary>
        ///  卡牌id
        /// </summary>
		[ProtoMember(5, IsRequired = false)]
		public string cardId;
        /// <summary>
        ///  摆放位置
        /// </summary>
		[ProtoMember(6, IsRequired = false)]
		public int position;
        /// <summary>
        ///  目标
        /// </summary>
		[ProtoMember(7, IsRequired = false)]
		public int target;
        /// <summary>
        ///  技能
        /// </summary>
		[ProtoMember(8, IsRequired = false)]
		public List<SkillInfo> skill;
        /// <summary>
        ///  是否成功
        /// </summary>
		[ProtoMember(1, IsRequired = false)]
		public bool success;
        /// <summary>
        ///  错误消息
        /// </summary>
		[ProtoMember(2, IsRequired = false)]
		public string info;

	}
}

File: trunk/server/src/main/java/org/ngame/protocol/battle/EjectionResponse.java
package org.ngame.protocol.battle;

import lombok.Data;

import java.util.List;

import org.ngame.protocol.BaseResponse;
import org.ngame.protocol.annotation.Proto;
import org.ngame.protocol.domain.SkillInfo;

import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;

/**
 *
 * @author beykery
 */
@Data
@Proto(value = -21, description = "出牌")
public class EjectionResponse extends BaseResponse
{
   @Protobuf(description = "用户id")
   private String pid;
   @Protobuf(description = "卡牌唯一id")
   private String uniqueId;
   @Protobuf(description = "卡牌id")
   private String cardId;
   @Protobuf(description = "摆放位置")
   private int position;
   @Protobuf(description = "目标")
   private int target;
   @Protobuf(description = "技能")
   private List<SkillInfo> skill;
}
